plasmapy/formulary/dimensionless.py

```diff
@@ -89,3 +89,71 @@ def beta(T: u.K, n: u.m ** -3, B: u.T) -> u.dimensionless_unscaled:
     thermal_pressure = parameters.thermal_pressure(T, n)
     magnetic_pressure = parameters.magnetic_pressure(B)
     return thermal_pressure / magnetic_pressure
+
+
+@validate_quantities(v={"can_be_negative": True})
+def Reynolds_number(rho: u.kg / u.m ** 3, v: u.m / u.s, L: u.m,
+                    mu: u.kg / (u.m * u.s)) -> u.dimensionless_unscaled:
+    """
+    The Reynold's Number is a dimensionless quantity
+    that is used to predict flow patterns in fluids.
+    The Reynold's Number is defined as the ratio of inertial forces to viscous forces.
+    A low Reynold's Number describes smooth, laminar flow
+    while a high Reynold's Number describes rough, turbulent flow.
+
+    .. math::
+
+        Re = \\rho v L / \\mu
+
+    Parameters
+    ----------
+    rho : `~astropy.units.Quantity`
+        The density of the plasma.
+    v : `~astropy.units.Quantity`
+        The flow velocity of the plasma.
+    L : `~astropy.units.Quantity`
+        The characteristic length scale.
+    mu : `~astropy.units.Quantity`
+        The viscosity of the plasma.
+
+     Warns
+    -----
+    ~astropy.units.UnitsWarning
+        If units are not provided, SI units are assumed.
+
+    Raises
+    ------
+    TypeError
+        The `v` is not a `~astropy.units.Quantity` and cannot be
+        converted into a ~astropy.units.Quantity.
+
+    ~astropy.units.UnitConversionError
+        If the `v` is not in appropriate units.
+
+    :exc:`~plasmapy.utils.exceptions.RelativityError`
+        If the velocity `v` is greater than the speed of light.
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> rho = 1000 * u.kg / u.m ** 3
+    >>> v = 10 * u.m / u.s
+    >>> L = 1 * u.m
+    >>> mu = 8.9e-4 * u.kg / (u.m * u.s)
+    >>> Reynolds_number(rho, v, L, mu)
+    <Quantity 11235955.05617978>
+    >>> rho = 1490 * u.kg / u.m ** 3
+    >>> v = 0.1 * u.m / u.s
+    >>> L = 0.05 * u.m
+    >>> mu = 10 * u.kg / (u.m * u.s)
+    >>> Reynolds_number(rho, v, L, mu)
+    <Quantity 0.745>
```

Comment:
As discussed during the telecon, adding docs.plasmapy.org/en/latest/api/plasmapy.formulary.braginskii.ion_viscosity.html?highlight=ion_viscosity#plasmapy.formulary.braginskii.ion_viscosity for mu could be a good call :)

Reply:
I guess we'll continue this discussion in #818 and close this for now.